lib/rules/no-assignment-of-untracked-properties-used-in-tracking-contexts.js

```diff
@@ -74,29 +158,64 @@ module.exports = {
       return {};
     }
 
+    const options = context.options[0] || { extraMacros: [] };
+
+    const macroConfigurations = [...DEFAULT_MACRO_CONFIGURATIONS, ...options.extraMacros];
+
+    const macrosByPath = new Map();
+    const macrosByIndexPath = new Map();
+
+    macroConfigurations.forEach((config) => {
+      if (!macrosByPath.has(config.path)) {
+        macrosByPath.set(config.path, new Map());
+      }
+      macrosByPath.get(config.path).set(config.name, config);
+
+      if (!macrosByIndexPath.has(config.indexPath)) {
+        macrosByIndexPath.set(config.indexPath, new Map());
+      }
+      if (!macrosByIndexPath.get(config.indexPath).has(config.indexName)) {
+        macrosByIndexPath.get(config.indexPath).set(config.indexName, new Map());
+      }
+      macrosByIndexPath.get(config.indexPath).get(config.indexName).set(config.name, config);
+    });
+
     // State being tracked for this file.
-    let computedImportName = undefined;
     let trackedImportName = undefined;
+    let computedImportName = undefined;
     let setImportName = undefined;
-    let macroImportNames = new Map();
+    let macrosByName = new Map();
+    let macrosByIndexName = new Map();
 
     // State being tracked for the current class we're inside.
     const classStack = new Stack();
 
     return {
       ImportDeclaration(node) {
+        const importPath = node.source.value;
+
+        if (macrosByPath.has(importPath)) {
```

Comment:
Could you add a comment with each of these top-level `if` statements inside this `ImportDeclaration` function to briefly explain what each case is looking for?